mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -80,21 +79,18 @@ impl<'de> serde::Deserialize<'de> for Settings {
 
         let upstream_connection_settings = nested_bridge
             .filter(|nested_bridge| {
-                nested_bridge.enable_upstream_bridge().map_or_else(
-                    || {
-                        debug!("upstream bridge not enabled.");
-                        false
-                    },
-                    |p| {
-                        debug!("upstream bridge setting {}", p);
-                        p
-                    },
-                )
+                nested_bridge
+                    .enable_upstream_bridge()
+                    .unwrap_or("false")
+                    .to_lowercase()
+                    == "true"
```

Comment:
looks weird.. does serde have problems converting to bool?

Reply:
it's not serde with the issue, it's environment from config, I looked at the code and it sets all env variables as string and that's why it can be deserialized properly. I wanted to write a workaround for that, but then acording to Varun doc there should be no new env variable added to enable bridge. I think it should be UpstreamProtocol which should be mqtt, but I will leave it like this for now until that's clarified